x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/search/transform_sort_field.ts

```diff
@@ -0,0 +1,83 @@
+import { assertUnreachable } from '../../../../../../common/utility_types';
+
+/**
+ * Transform the sort field name from the request to the Alerting framework
+ * schema.
+ *
+ * It could be a bit confusing what field is used for sorting rules as it is
+ * hidden under several layers of abstraction.
+ *
+ * Sort field as defined in the UI rule schema -> API transformation (this
+ * function) -> Alerting framework transformation -> Saved Object Client
+ * transformation -> Elasticsearch field
+ *
+ * The alerting framework applies the following transformations to the sort
+ * field:
+ * - Some sort fields like `name` are converted to `name.keyword` by the
+ *   Alerting framework automatically
+ * - Sort fields that have corresponding mapped params, like `riskScore`, will
+ *   be prefixed with `mapped_params.` and converted to a snake case
+ *   automatically. So `riskScore` will become `mapped_params.risk_score`
+ * - Other fields will be passed to the saved object client as is.
+ *
+ * Saved objects client also applies some transformations to the sort field:
+ * - First, it appends the saved object type to the sort field. So `name` will
+ *   become `alert.name`
+ * - If the sort field doesn't exist in saved object mappings, then it tries the
+ *   field without prefixes, e.g., just `name`.
+ *
+ * @param sortField Sort field parameter from the request
+ * @returns Sort field matching the Alerting framework schema
+ */
+export function transformSortField(sortField: SortFieldOrUndefined): string | undefined {
+  if (!sortField) {
+    return undefined;
+  }
+
+  switch (sortField) {
+    // Without this conversion, rules will be sorted by the top-level SO fields.
+    // That is seldom what is expected from sorting. After conversion, the
+    // fields will be treated as `alert.updatedAt` and `alert.createdAt`
+    case 'created_at':
+      return 'createdAt';
+    case 'updated_at':
+      return 'updatedAt';
+
+    // Convert front-end representation to the field names that match the Alerting framework mappings
+    case 'execution_summary.last_execution.date':
+      return 'monitoring.run.last_run.timestamp';
+    case 'execution_summary.last_execution.metrics.execution_gap_duration_s':
+      return 'monitoring.run.last_run.metrics.gap_duration_s';
+    case 'execution_summary.last_execution.metrics.total_indexing_duration_ms':
+      return 'monitoring.run.last_run.metrics.total_indexing_duration_ms';
+    case 'execution_summary.last_execution.metrics.total_search_duration_ms':
+      return 'monitoring.run.last_run.metrics.total_search_duration_ms';
+    case 'execution_summary.last_execution.status':
+      return 'lastRun.outcomeOrder';
+
+    // Pass these fields as is. They will be converted to `alert.<field>` by the saved object client
+    case 'createdAt':
+    case 'updatedAt':
+    case 'enabled':
+    case 'name':
+    case 'version':
```

Comment:
Seeing `unknown sort field` when sorting by `version`:

<img width="340" alt="image" src="https://user-images.githubusercontent.com/2946766/216478985-ec416776-db03-4534-a49e-51874f63ffa1.png">


Debugging at the SO layer shows `version` exists in the `security-rule` mapping:

<img width="387" alt="image" src="https://user-images.githubusercontent.com/2946766/216478516-87775f1f-5b5d-4ee3-8a99-df82642d83f2.png">

It just fails to look it up based off the base `version` sortField. If you do an inline replace with `security-rule.version` it'll find the right field but still won't work. We'd need `version` added as a `mapped_param` so it's indexed/sortable, no? Though not sure why it's showing up in that mapping along with `name`/`rule_id`...?

---

We're planning on removing the `Version` column anyway as part of https://github.com/elastic/kibana/issues/148196, so no need to mess with any of this most likely.

Reply:
It's a nice catch! We should remove `version` from the allowed sort fields and make that column not sortable. And then delete it completely as part of https://github.com/elastic/kibana/issues/148196.

Reply:
I decided to remove the version field in this PR to save some time opening another PR with just the column removal 😅